trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py

```diff
@@ -0,0 +1,338 @@
+        self.training_file             = None
+        
+        # Surrogate model
+        self.surrogates = Data()
+        self.surrogates.moment_coefficient = None
+
+    def initialize(self):
+
+        geometry = self.geometry
+        self.tag      = 'avl_analysis_of_{}'.format(geometry.tag)
+
+        run_folder = self.settings.filenames.run_folder
+        #if os.path.exists(run_folder):
+            #if self.keep_files:
+                #warn('deleting old avl run files',Warning)
+            #rmtree(run_folder)
+        #os.mkdir(run_folder)
+        
+        # Sample training data
+        self.sample_training()
+    
+        # Build surrogate
+        self.build_surrogate()
+
+
+        return
+
+    def evaluate(self,state,settings,geometry):
+
+        # Unpack
+        surrogates = self.surrogates        
+        conditions = state.conditions
+        
+        mach = conditions.freestream.mach_number
+        AoA  = conditions.aerodynamics.angle_of_attack
+        lift_model = surrogates.lift_coefficient
+        drag_model = surrogates.drag_coefficient
+        moment_model = surrogates.moment_coefficient
+        
+        # Inviscid lift
+        data_len = len(AoA)
+        inviscid_lift = np.zeros([data_len,1])
+        for ii,_ in enumerate(AoA):
+            inviscid_lift[ii] = lift_model.predict(np.array([AoA[ii][0],mach[ii][0]]))
+        conditions.aerodynamics.lift_breakdown.inviscid_wings_lift = inviscid_lift
+        state.conditions.aerodynamics.lift_coefficient             = inviscid_lift
+        state.conditions.aerodynamics.lift_breakdown.compressible_wings = inviscid_lift
+        
+        # Inviscid drag, zeros are a placeholder for possible future implementation
+        inviscid_drag = np.zeros([data_len,1])        
+        state.conditions.aerodynamics.inviscid_drag_coefficient    = inviscid_drag
+        
+        return inviscid_lift, inviscid_drag
+       
+       
+        
+
+    def sample_training(self):
+        
+        # Unpack
+        geometry = self.geometry
+        training = self.training
+        
+        AoA  = training.angle_of_attack
+        mach = training.Mach
+        
+        CL   = np.zeros([len(AoA)*len(mach),1])
+        CD   = np.zeros([len(AoA)*len(mach),1])
+        CM   = np.zeros([len(AoA)*len(mach),1])
+
+        
+        if self.training_file is None:
+            # Calculate aerodynamics for table
+            table_size = len(AoA)*len(mach)
+            xy = np.zeros([table_size,2])
+            count = 0
+            time0 = time.time()
+            
+            for i,_ in enumerate(mach):
+                for j,_ in enumerate(AoA):
+                    xy[i*len(mach)+j,:] = np.array([AoA[j],mach[i]])
+            for j,_ in enumerate(mach):
+                # Set training conditions
+
+                run_conditions = Aerodynamics()
+                run_conditions.weights.total_mass           = geometry.mass_properties.max_takeoff
```

Comment:
Not currently however it will become useful in the future when we tap into more capabilities of AVL